client.go

```diff
@@ -171,6 +174,7 @@ func (c *Client[Req, Res]) newConn(ctx context.Context, streamType StreamType) S
 }
 
 type clientConfig struct {
+	URL                    *url.URL
```

Comment:
I couldn't tell anywhere clientConfig or any way this got exposed even though the fields are suggested to be public, so this felt safe to do. It seems this is only used internally, and most of the other fields on this struct get reused to be passed along to the `protocolClientParams` too.